src/orders/components/OrderHistory/OrderHistory.tsx

```diff
@@ -91,32 +135,57 @@ const OrderHistory = ({
         {history ? (
           <Timeline>
             <Form confirmLeave initial={{ message: "" }} onSubmit={onNoteAdd} resetOnSubmit>
-              {({ change, data, reset, submit }) => (
+              {({ change, data, reset, submit, triggerChange }) => (
                 <TimelineAddNote
                   message={data.message}
                   reset={reset}
-                  onChange={change}
+                  onChange={e => {
+                    change(e);
+
+                    // Reset dirty state if message is empty (back to initial state)
+                    if (e.target.value === "") {
+                      triggerChange(false);
+                    }
+                  }}
                   onSubmit={submit}
+                  label={intl.formatMessage({
+                    id: "LgbKvU",
+                    defaultMessage: "Comment",
+                  })}
+                  buttonLabel={
+                    <FormattedMessage
+                      id="H5jL5+"
+                      defaultMessage="Add Comment"
+                      description="button"
+                    />
+                  }
                 />
               )}
             </Form>
-            {history
-              .slice()
-              .reverse()
-              .map(event => {
+            {(() => {
+              const reversedHistory = history.slice().reverse();
+              const groupedEvents = groupEventsByDate(reversedHistory);
```

Comment:
suggestion: let's use `useMemo` for this - right now we'll have to calculate it on every re-render, it can get problematic if we have a lot of order events